components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3669,6 +3669,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     async getIDToken(): Promise<void> {}
     public async resolveContext(ctx: TraceContextWithSpan, contextUrl: string): Promise<WorkspaceContext> {
         const user = this.checkAndBlockUser("resolveContext");
-        return this.contextParser.handle(ctx, user, contextUrl);
+        const normalizedCtxURL = this.contextParser.normalizeContextURL(contextUrl);
```

Comment:
Is there a time when you wouldn't want to normalize? And if no, can we move the normalization inside the `contextParser.handle` method?

Reply:
There are places where we don't do the normalization, which is when contexts are created based on git hoster events, where we don't need to be fuzzy around the allowed input. It would probably not harm to call the normalization in those cases as well, but I would like to avoid changing the implementation of contextParser with this PR. It's a bit complex and I don't see enough value in the proposed change.